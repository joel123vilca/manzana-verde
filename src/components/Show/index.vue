<template>
  <div class="detail">
    <div class="detail__head">
      <a @click="$router.go(-1)" class="back__button">Atras</a>
      <h2 class="title">Almuerzo</h2>
      <div class="date">
        <span class="date__day">
          Vie
        </span>
        <span class="date__number">
          27
        </span>
      </div>
    </div>
    <div class="detail__foods">
      <div v-for="(food, index) in foods" :key="index" class="food">
        <div class="food__image">
          <img :src="require('@/assets/food.png')" alt="food" />
        </div>
        <div class="food__info" @click="$router.push({ name: 'Detail', params: { id: food.id } })">
          <h3 class="food__title">{{ food.name }}</h3>
          <button class="recomend">Recomendado</button>
        </div>
        <div class="food__description">
          <div class="count">{{ food.kcal }} kcal</div>
          <div class="icons">
            <img
              v-for="item in food.detail"
              :key="item.name"
              :src="require(`@/assets/${item.icon}`)"
              alt="sugar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script src="./show.js"></script>
<style src="./show.scss" scoped lang="scss"></style>
