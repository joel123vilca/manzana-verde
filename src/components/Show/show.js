import { mapState } from "vuex";
export default {
  data() {
    return {
      isActive: true
    };
  },
  computed: {
    ...mapState({
      foods: state => state.foods
    })
  }
};
