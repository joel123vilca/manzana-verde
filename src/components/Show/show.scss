.detail {
  .detail__head {
    height: 60px;
    border-bottom: 2px solid #e8e8e8;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .title {
      size: 20px;
      color: #606060;
    }
    .back__button {
      color: #6b757d;
      size: 14px;
      font-weight: 600;
    }
    .date {
      display: flex;
      flex-direction: column;
      .date__number {
        font-weight: 700;
        size: 20px;
        color: #606060;
      }
    }
  }
  .detail__foods {
    padding: 10px;
    .food {
      margin-top: 20px;
      border: 1px solid #e8e8e8;
      border-radius: 10px;
      .food__image {
        img {
          border-radius: 10px 10px 0 0;
          width: 100%;
          object-fit: contain;
        }
      }
      .food__info {
        display: flex;
        justify-content: space-between;
        align-items: center;
        cursor: pointer;
        .food__title {
          color: #35b266;
          text-align: left;
          font-size: 17px;
          margin-left: 10px;
        }
        .recomend {
          color: #35b266;
          border: 1px solid rgba(53, 178, 102, 0.1);
          background-color: rgba(53, 178, 102, 0.1);
          border-radius: 5px;
          width: 120px;
          height: 30px;
          margin-right: 10px;
        }
      }
      .food__description {
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding-bottom: 10px;
        cursor: pointer;
        .count {
          margin-left: 10px;
        }
        .icons {
          margin-right: 10px;
          display: flex;
          img {
            margin-left: 15px;
          }
        }
      }
    }
  }
}
@media (min-width: 992px) {
  .detail__foods {
    display: grid;
    grid-template-columns: 45% 45%;
    grid-gap: 50px;
  }
}
