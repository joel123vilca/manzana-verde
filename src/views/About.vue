<template>
  <div>
    <Show />
  </div>
</template>
<script>
import Show from "@/components/Show";
export default {
  components: {
    Show
  }
};
</script>
