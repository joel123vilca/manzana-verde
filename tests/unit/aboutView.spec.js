import { shallowMount } from "@vue/test-utils";
import About from "@/views/About";
import Show from "@/components/Show";

describe("about view", () => {
  const build = () => {
    const wrapper = shallowMount(About);

    return {
      wrapper,
      Show: () => wrapper.find(Show)
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the right", () => {
    const { Show } = build();
    expect(Show().exists()).toBe(true);
  });
});
