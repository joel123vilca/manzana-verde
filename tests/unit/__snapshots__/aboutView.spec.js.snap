// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about view renders the component 1`] = `
<div>
  <show-stub></show-stub>
</div>
`;
